homeassistant/components/recorder/__init__.py

```diff
@@ -554,3 +550,34 @@ def _close_run(self):
             self.event_session.close()
 
         self.run_info = None
+
+
+def engine_args_for_db_url(db_url):
+    """Build the engine args for given db url."""
+    if db_url == "sqlite://" or ":memory:" in db_url:
+        return {
+            "connect_args": {"check_same_thread": False},
+            "poolclass": StaticPool,
+            "pool_reset_on_return": None,
+        }
+
+    db_is_sqlite = db_url.startswith("sqlite://")
+
+    if db_is_sqlite:
+        import sqlite3  # pylint: disable=import-outside-toplevel
+
+    if not db_is_sqlite or not sqlite3.threadsafety:
+        return {"echo": False}
+
+    # https://www.sqlite.org/threadsafe.html
+    #
+    # In serialized mode, SQLite can be safely used
+    # by multiple threads with no restriction.
+    #
+    # The default mode is serialized.
+    return {
+        "echo": False,
+        "connect_args": {"check_same_thread": False},
+        "poolclass": QueuePool,
+        "pool_timeout": SQLITE_POOL_TIMEOUT,
+    }
```

Comment:
I think the fallback logic here makes it a bit hard to read.

Wouldn't this be easier to understand (unless i'm missing something):
```suggestion
    # https://www.sqlite.org/threadsafe.html
    #
    # In serialized mode, SQLite can be safely used
    # by multiple threads with no restriction.
    #
    # The default mode is serialized.
    if db_url.startswith("sqlite://"):
        import sqllite3
        if sqllite3.theadsafety:
            return {
                "echo": False,
                "connect_args": {"check_same_thread": False},
                "poolclass": QueuePool,
                "pool_timeout": SQLITE_POOL_TIMEOUT,
            }

    return {"echo": False}
```

Reply:
adjusted as suggested